File: time2secs/time2secs.rb
#!/usr/bin/env ruby

def time2secs(time)
    secs = [1, 60, 3600]
    time = time.split(":").reverse
    seconds = 0

    for i in 0..time.length-1
        seconds +=  secs[i] * time[i].to_i
    end

    return "=== " + seconds.to_s + " seconds ==="
end

if ARGV.length == 1
    puts time2secs(ARGV[0])
else
    puts "Please enter a command line arg"
    puts "HH:MM:SS"
    puts "EX:"
    puts "    time2secs 12:34:56"
end

File: time2secs/time2secs.py
#!/usr/bin/python

import sys

def time_to_seconds(time):
    time = [ int(i) for i in time.split(":") ]
    return sum( [ [1,60,3600][i]*j for i, j in enumerate(reversed(time[:3]))] )

if __name__ == "__main__":
    if len(sys.argv) == 2:
        print "=== %s seconds ===" % time_to_seconds(sys.argv[1])
    else:
        print "Please enter a command line arg"
        print "HH:MM:SS"
        print "EX:"
        print "    time2secs 12:34:56"

